examples/ecs/state_scoped.rs

```diff
@@ -0,0 +1,123 @@
+//! Shows how to use state scoped systems.
```

Comment:
If we're keeping this example, the module docs here need more color on what this feature is and why it's useful.

I'd probably prefer to cut it, in favor of the already solid doc tests. Just make sure that the state scoped stuff is linked in the module docs of `bevy_state`.